lib/rucio/client/downloadclient.py

```diff
@@ -288,7 +289,7 @@ def download_rucio(self, pfn, protocol, input_queue, output_queue, trace_pattern
                 thread.kill_received = True
         logger.debug('All threads finished')
 
-    def _downloader(self, pfn, protocol, human, input_queue, output_queue, user_agent, threadnb, total_threads, trace_endpoint, trace_pattern):
+    def _downloader(self, pfn, protocol, human, input_queue, output_queue, user_agent, threadnb, total_threads, trace_endpoint, trace_pattern, transfer_timeout=3600):
```

Comment:
Why is the timeout `3600` in some places and `None` at others?

Seems that if the user doesn't specify one, we should ask the configuration for a timeout rather than scatter a "magic integer" through the codebase.

Reply:
The ```None``` timeout is just a placeholder for the protocols where no timeout is implemented yet.

Where the protocol is implemented, I selected the magic value of ```3600``` just because it's what is hardcoded in the gfal protocol in the current production version, but of course I'm all for more sensible suggestions.

Reply:
Oh, I'm not sure I have more sensible suggestion of a value -- I just want to aggregate it in one place instead of distributing it throughout the codebase.

Reply:
I'm wondering if it's not better to extend the protocols `get` and `put` functions with **kwargs instead in the signature. It's more generic and allow more possible extensions

Reply:
I think it should be explicit, wherever possible, as it increases the readability a lot. But when we have a usecase for a protocol we might want to add the **kwargs on top of that.